actions_helper/main.py

```diff
@@ -1,72 +1,102 @@
+            application_id=f"{ecr_repository}-local-exec-{environment}",
+            deployment_tag=deployment_tag,
+            image_uri=image_uri,
+        )
 
-@cli.command(name="wait-for-task-stopped")
-@click.option("--aws-region", default=os.environ.get("AWS_DEFAULT_REGION"), type=str)
-@click.option("--cluster", type=str)
-@click.option("--task", type=str)
-def cmd_wait_for_task_stopped(
-    aws_region: str,
-    cluster: str,
-    task: str,
-):
-    wait_for_task_stopped(
-        ecs_client=boto3.Session(region_name=aws_region).client("ecs"),
-        cluster=cluster,
-        task=task,
-    )
+        click.echo("Creating production task definition...")
+        production_task_definition = create_task_definition(
+            ecs_client=ecs_client,
+            application_id=f"{ecr_repository}-{environment}",
+            deployment_tag=deployment_tag,
+            image_uri=image_uri,
+        )
 
+        if run_preflight:
+            click.echo("Run preflight enabled")
+            preflight_output = run_preflight_container(
+                ecs_client=ecs_client,
+                service=f"{ecr_repository}-{environment}",
+                cluster=environment,
+                application_id=f"{ecr_repository}-preflight-{environment}",
+                image_uri=image_uri,
+                deployment_tag=deployment_tag,
+            )
 
-@cli.command(
-    name="get-active-task-definition-arn-by-tag",
-    short_help="Get active task definition ARN  by specified tags",
-)
-@click.option("--application-id", default=os.environ.get("APPLICATION_ID"), type=str)
-@click.option("--tags", default=os.environ.get("TAGS"), type=str)
-@click.option("--aws-region", default=os.environ.get("AWS_DEFAULT_REGION"), type=str)
-@click.option("--allow-initial-deployment", is_flag=True, default=False)
-def cmd_get_active_task_definition_arn_by_tag(
-    application_id: str,
-    tags: str,
-    aws_region: str,
-    allow_initial_deployment: bool,
-):
-    click.echo(
-        get_active_task_definition_arn_by_tag(
-            ecs_client=boto3.Session(region_name=aws_region).client("ecs"),
-            task_definition_family_prefix=application_id,
-            task_definition_tags=tags,
-            allow_initial_deployment=allow_initial_deployment,
-        ),
-    )
+        click.echo("Updating service...")
+        ecs_client.update_service(
+            taskDefinition=production_task_definition.latest_task_definition_arn,
+            desiredCount=desired_count,
+            cluster=environment,
+            service=f"{ecr_repository}-{environment}",
+        )
+
+        click.echo("Service updated")
+
+        click.echo("Waiting for service stability...")
+        wait_for_service_stability(
+            ecs_client=ecs_client,
+            cluster=environment,
+            service=f"{ecr_repository}-{environment}",
+        )
+
+    except Exception as e:
+        click.echo("An exception occurred", err=True)
+        click.echo(e)
```

Comment:
Can you explain your thoughts here? What additional value does the `except` part provide? From my understanding, we could simply drop the `except` block, as `finally` would still execute before raising the exception.

Reply:
`finally` is executed after the raising the exception
The `except` part is useful to see the stack trace of the thrown error (if any)

Reply:
We have talked about this and could confirm that the `expert` part can be removed.